thing-management.md

```diff
@@ -1,8 +1,356 @@
 
-## Adding a thing to FH system
+## Adding a thing to FH system - SmartStart (Z-Wave only)
+
+SmartStart inclusion is a process of adding a device with a pre-configuration step using a DSK printed on the back of the device. With this functionality, you simply set up your device and power it on. The device is then automatically recognized and added to the network because it already comes with a configuration set by default.
+
+There are two ways of adding a device to the NPL:
+
+1. Scanning the DSK using a mobile phone camera (QR).
+2. Manually entering the DSK into the Node Provisioning List.
+
+### What is the Node Provisioning List?
+
+NPL is a list which contains all the DSKs for devices that will be added using SmartStart (if those are SmartStart capable) or by classic inclusion with Security S2 without the additional classic inclusion security bootstrapping key exchange step. It is a Z-Wave abstraction which is used mostly for SmartStart functionality but also for simplifying inclusion of S2 capable nodes. It comes with the possibility to add, edit or remove its entries.
+
+### Additional documentation for SmartStart and NPL
+
+All documents are located in z-wave/smartstart directory.
+
+1. Node Provisioning Information Type Registry: Most of properties used in FIMP model are taken from this document. Example: 3.1.2.8 SmartStart Inclusion Setting Information Type is basically translated to "inclusion_setting" in FIMP protocol.
+2. Node Provisioning QR Code Format: Document for any kind of front end application which would want to implement QR Code scanning capability.
+3. Z-Wave Network Protocol Command Class Specification: Adnotation 4.5.12.5 shows diagrams for use cases of SmartStart and NPL.
+
+### NPL Data Model
+
+All of the properties are optional with three exceptions: 'inclusion_setting', 'bootstraping_mode' and 'network_status'.
+
+#### `inclusion_setting`
+
+Can be one of the following:
+
+- `pending`: Device will be added to network right after powering it up. 
+- `passive`: Controller has decided to not perform inclusion right after powering up a device. When get for all entries is executed this state is changed automatically to 'pending'.
+- `ignored`: Controller has decided to not perform inclusion right after powering up a device. This state can be changed only manually.
+
+#### `bootstrapping_mode`
+
+Can be one of the following:
+
+- `s2_only`: Device must manually be set to Learn Mode and follow S2 bootstraping instructions.
+- `smart_start`: Device will be included and S2 bootstrapped automatically using Z-Wave Smart Start.
+- `smart_start_lr`: Device will be included and S2 bootstrapped automatically using Z-Wave Long Range Smart Start.
+
+#### `network_status`
+
+Object containing nodeId and status.
+
+> `status` can be `not_in_network`, `failing` or `included`.
```

Comment:
Can you add info about the different statuses here?

Reply:
done